database-commons/src/main/java/io/cdap/plugin/db/config/AbstractDBSpecificSourceConfig.java

```diff
@@ -125,16 +148,34 @@ public void validate(FailureCollector collector) {
       TransactionIsolationLevel.validate(getTransactionIsolationLevel(), collector);
     }
 
-    if (!containsMacro(IMPORT_QUERY) && Strings.isNullOrEmpty(importQuery)) {
-      collector.addFailure("Import Query is empty.", "Specify the Import Query.")
-        .withConfigProperty(IMPORT_QUERY);
+    if (!containsMacro(PROPERTY_IMPORT_QUERY_TYPE)) {
+      ImportQueryType importQueryType = ImportQueryType.fromString(getImportQueryType());
+
+      boolean isImportQuerySelected = importQueryType == ImportQueryType.IMPORT_QUERY;
```

Comment:
These conditions are already tested in redshift and postgres, here add just a single condition if both values are null or empty or macro. keep just the one on line 166